### PR TITLE
Fix add departments

### PR DESCRIPTION
**Pull request by VillyGH**

### CODE REVIEW

File: src/Constants/testConstants.ts [hunk 1]
```diff
@@ -22,57 +22,59 @@ export const testPopupConstant = {
...
+    id: "1234",
     firstName: "John",
     lastName: "Trenta",
     email: "jtrenta@hotmail.com",
-    isActive: true,
     phoneNumber: "976-433-1062",
     department: "Marketing",
     jobTitles: ["Manager"],
     skills: ["Gestion de personnel"],
+    isActive: false,
```
Comment (BlobMaster41) [inline src/Constants/testConstants.ts:75]: false? change to true

File: src/src/engine/api/APIManager.ts [hunk 2]
```diff
@@ -285,7 +283,7 @@ class APIManager extends Logger {
      * @memberof APIManager
      */
     private requestUserCreationFromWorker(
-        employee: EmployeeCreateDTO,
+        employee: Employee,
```
Comment (DJLawton) [inline src/src/engine/api/APIManager.ts:286]: Explain, plz

**Review by BlobMaster41 — CHANGES_REQUESTED** [review]

File: src/src/engine/components/ComponentAddEmployee.tsx [hunk 3]
```diff
@@ -252,11 +253,13 @@ export class ComponentAddEmployee extends React.Component<AddEmployeeProps, Comp
...
                 department: this.state.department,
+                isActive: this.state.isActive,
                 jobTitles: this.state.jobTitles,
                 skills: this.state.skills, // @ts-ignore
                 role: parseInt(this.state.role),
-                hasChangedDefaultPassword: false
-            }
+                hasChangedDefaultPassword: this.state.hasChangedDefaultPassword
+            };
+            console.log(employee);
```
Comment (DJLawton) [inline src/src/engine/components/ComponentAddEmployee.tsx:262]: remove console.log

File: src/src/engine/pages/AddEmployee.tsx [hunk 4]
```diff
@@ -90,7 +90,7 @@ export class AddEmployee extends React.Component<unknown, AddEmployeeState> {
     /**
      * Add an employee to the database
      * @param password {string} The password of the employee
-     * @param employee {EmployeeCreateDTO} The employee to add
+     * @param employee {Employee} The employee to add
      * @private
      */
     readonly #addEmployee = async (password: string, employee: EmployeeCreateDTO): Promise<void> => {
```
Comment (DJLawton) [inline src/src/engine/pages/AddEmployee.tsx:96]: straight up lies

File: src/src/engine/components/ComponentEditEmployee.tsx [hunk 5]
```diff
@@ -46,6 +50,15 @@ export class ComponentEditEmployee extends React.Component<EditEmployeeProps, Co
         this.props = props;
     }
 
+    componentDidUpdate(prevProps: EditEmployeeProps) {
```
Comment (BlobMaster41) [inline src/src/engine/components/ComponentEditEmployee.tsx:53]: public : void

**Review by BlobMaster41 — CHANGES_REQUESTED** [review]

**Pull request merged**